harness/determined/pytorch/_reducer.py

```diff
@@ -183,8 +184,25 @@ def update(self, value: Any) -> None:
     def per_slot_reduce(self) -> Any:
         return self.values
 
+    class _NotMetric:
+        pass
+
+    NOT_METRIC = _NotMetric()
+
     def cross_slot_reduce(self, per_slot_metrics: List) -> Any:
-        flat_metrics = [item for sublist in per_slot_metrics for item in sublist]
+        """
+        Call the reducer function on metrics in their original order.
+
+        Interleave metrics from different slots as we flatten
```

Comment:
No need to reformat, but fwiw I don't think we normally wrap docstring lines this narrowly.

If there is good reason to start, let me know though.

Reply:
No, there is no good reason to go down from 100; we have ample real estate nowadays.

The first sentence is the "header"; it is followed by an empty line.
The second sentence does not fit into one line, and rather than split it closer to the 100 character limit I prefer to split it in the middle.
The third sentence starts with "Note:" and therefore deserves to be on a separate line IMHO. And again it does not fit on a single line.